### ISSUE TITLE
[Feature Request] Add new check to validate undefined variable

### ISSUE THREAD

**Issue by moylop260** [PyCQA/pylint#1376]
### Steps to reproduce
1. Running the following code
```python
if False:
    new_variable = "hello"
print new_variable
```

We have the following error:
```python
Traceback (most recent call last):
  File "example.py", line 3, in <module>
    print new_variable
NameError: name 'new_variable' is not defined
```

### Current behavior
If we run a pylint to check if this error is detected early we don't have a check for this case.

### Expected behavior
Emit a check similar to `W0631 undefined-loop-variable - Using possibly undefined loop variable 'new_variable'` but for the `if` section.

### pylint --version output
Latest master sha for astroid and pylint

**Comment by degustaf** [comment]
Duplicate of #85.
